### PR TITLE
Synchronize HIDTables.h with Kaleidoscope

### PR DESCRIPTION
**Pull request by gedankenexperimenter**
This updates the USB HID Usage Tables to version 1.2 (see keyboardio/Kaleidoscope#901)

### CODE REVIEW

File: src/HIDTables.h [hunk 1]
```diff
@@ -48,655 +39,762 @@ THE SOFTWARE.
...
-#define HID_CONSUMER_AC_SYNCHRONIZE	0x286	// HID type SEL
-#define HID_CONSUMER_AC_SEND_SLASH_RECEIVE	0x287	// HID type SEL
-#define HID_CONSUMER_AC_SEND_TO	0x288	// HID type SEL
-#define HID_CONSUMER_AC_REPLY	0x289	// HID type SEL
-#define HID_CONSUMER_AC_REPLY_ALL	0x28A	// HID type SEL
-#define HID_CONSUMER_AC_FORWARD_MSG	0x28B	// HID type SEL
-#define HID_CONSUMER_AC_SEND	0x28C	// HID type SEL
-#define HID_CONSUMER_AC_ATTACH_FILE	0x28D	// HID type SEL
-#define HID_CONSUMER_AC_UPLOAD	0x28E	// HID type SEL
-#define HID_CONSUMER_AC_DOWNLOAD_(SAVE_TARGET_AS)	0x28F	// HID type SEL
```
Comment (gedankenexperimenter) [inline src/HIDTables.h:687]: Incidentally, this PR also fixes an old bug where this keycode macro was misnamed. This was fixed in Kaleidoscope quite a while ago.

**Comment by obra** [comment]
Thanks for this. I'm going to end up rewriting the commits because there's an unintentional license change in here.

**Comment by obra** [comment]
Thanks. Merged an edited version of the commit (which only changes comments)

**Comment by gedankenexperimenter** [comment]
> Thanks for this. I'm going to end up rewriting the commits because there's an unintentional license change in here.

Whoops! Sorry about that.

**Pull request closed without merging**